Aztec/Classes/TextKit/TextView.swift

```diff
@@ -724,6 +767,56 @@ open class TextView: UITextView {
     }
 
 
+    /// Removes the List Attributes from a collection of attributes, whenever:
+    ///
+    ///     A. The caret is at the very end of the document (+ there is no selected text)
+    ///     B. There's a list!
+    ///     C. The previous character is a '\n'
+    ///
+    /// - Parameter attributes: Typing Attributes.
+    ///
+    /// - Returns: Updated Typing Attributes.
+    ///
+    private func ensureRemovalOfListAttribute(from attributes: [String: Any]) -> [String: Any] {
+        guard selectedRange.location == storage.length && selectedRange.length == 0 else {
```

Comment:
For reading simplicity I'd consider removing `selectedRange.length == 0`.  If the caret position is indeed at EOF, then the length will always be zero.